src/main/java/net/dv8tion/jda/api/entities/ThreadChannel.java

```diff
@@ -70,34 +136,81 @@ default GuildMessageChannel getParentMessageChannel()
         throw new UnsupportedOperationException("Parent of this thread is not a MessageChannel. Parent is type: " + getParentChannel().getType().getId());
     }
 
+    /**
+     * Gets the self member, as a member of this thread.
+     *
+     * <br>If the current account is not a member of this thread, this will return null.
+     *
+     * @return The self member of this thread, null if the current account is not a member of this thread.
+     *
+     * @see #isJoined()
+     */
     @Nullable
     default ThreadMember getSelfThreadMember()
     {
         return getThreadMember(getJDA().getSelfUser());
     }
 
-    //Only have access to this with GUILD_MEMBERS
+
+    /**
+     * Gets a List of all cached {@link ThreadMember members} of this thread.
+     * <br>
+     * <br>The thread owner is not included in this list.
+     * Any updates to this cache are lost when JDA is shutdown, and this list is not sent to JDA on startup.
+     * For this reason, {@link #retrieveThreadMembers()} should be used instead in most cases.
+     *
+     * <p>In order for the cache to be updated, the following requirements must be met:
+     * <ul>
+     *     <li>the {@link net.dv8tion.jda.api.requests.GatewayIntent#GUILD_MEMBERS} intent must be enabled.</li>
+     *     <li>the bot must be able to join the thread (either via the {@link net.dv8tion.jda.api.Permission#MANAGE_THREADS MANAGE_THREADS} permission, or a public thread)</li>
+     *     <li>the bot must have be online to receive the update</li>
+     * </ul>
+     *
+     * @return a List of all {@link ThreadMember members} of this thread. This list may be empty, but not null.
+     *
+     * @see #retrieveThreadMembers()
+     */
```

Comment:
The documentation here is pretty good, but it needs to be drilled down that this list is _basically empty_ until we get thread member join events. That's the only thing that populates this array. and ofc, they are lost on bot restart.